core/common/src/main/java/alluxio/security/authorization/AclEntry.java

```diff
@@ -199,20 +257,38 @@ public static AclEntry fromCliStringWithoutPermissions(String stringEntry) {
     }
     List<String> components = Arrays.stream(stringEntry.split(":")).map(String::trim).collect(
         Collectors.toList());
-    if (components.size() < 1 || components.size() > 3) {
+    if (components.size() < 1 || components.size() > 4) {
       throw new IllegalArgumentException("Unexpected acl components: " + stringEntry);
     }
 
     AclEntry.Builder builder = new AclEntry.Builder();
-    String type = components.get(0);
-    String subject = "";
-    if (components.size() >= 2) {
-      subject = components.get(1);
-    }
-    String actions = "";
-    if (components.size() >= 3) {
-      actions = components.get(2);
+    String type;
+    String subject;
+    String actions;
+    if (components.get(0).equals("default")) {
```

Comment:
can we do this the `startingIndex` way like above?

Reply:
i feel using `startIndex` is more error prone for more complicated logic like this method.